master/internal/api_workspace.go

```diff
@@ -142,15 +144,17 @@ func (a *apiServer) validateClusterNamespaceMeta(
 		// This might occur when using singleRM with a cluster name defined in the master config,
 		// but not specified in a given request.
 		if newClusterName != clusterName {
-			namespaceMeta[newClusterName] = &workspacev1.WorkspaceNamespaceMeta{
-				ClusterName: clusterName,
-				Namespace:   namespace,
+			namespaceMetaWithAllClusterNames[newClusterName] = &workspacev1.WorkspaceNamespaceMeta{
+				ClusterName:         clusterName,
```

Comment:
Should this be `newClusterName` instead of `clusterName` here? I guess I'm a little confused that there's a map where the key is `newClusterName` but the value has a field where `ClusterName = clusterName`.

Reply:
Awesome catch! Yea it should be `newClusterName`. Thankss